pkg/web/apps/apps_service_test.go

```diff
@@ -85,3 +87,238 @@ func Test_appsService_GetAppByID(t *testing.T) {
+					DeviceID:      "3ee2a7e7-25e4-465f-a998-c781b3cb9d46",
+					Version:       "2.0",
+					DeviceVersion: "8.1",
+					DeviceType:    "Android",
+				},
+			},
+		},
+		{
+			name:    "InitClient() should create a new device when device with device ID and app ID not found",
+			wantErr: nil,
+			args: args{
+				deviceInfo: &models.Device{
+					ID:            "ccf2c998-992e-4db4-b395-22ac515d6c21",
+					VersionID:     "12da0a46-f94d-4ab3-8e3e-f4366d63d14d",
+					AppID:         "com.aerogear.testapp",
+					DeviceID:      "3ee2a7e7-25e4-465f-a998-c781b3cb9d46",
+					Version:       "2.0",
+					DeviceVersion: "8.1",
+					DeviceType:    "Android",
+				},
+			},
+			fields: fields{
+				app: &models.App{
+					ID:      uuid.New().String(),
+					AppID:   "com.aerogear.testapp",
+					AppName: "Test App",
+				},
+				version: &models.Version{
+					ID:                   uuid.New().String(),
+					AppID:                "com.aerogear.testapp",
+					Version:              "2.0",
+					Disabled:             false,
+					DisabledMessage:      "",
+					NumOfCurrentInstalls: 1000,
+					NumOfAppLaunches:     10000,
+				},
+				device: &models.Device{
+					ID:            "ccf2c998-992e-4db4-b395-22ac515d6c21",
+					VersionID:     "12da0a46-f94d-4ab3-8e3e-f4366d63d14d",
+					AppID:         "com.aerogear.testapp",
+					DeviceID:      "3ee2a7e7-25e4-465f-a998-c781b3cb9d46",
+					Version:       "2.0",
+					DeviceVersion: "8.1",
+					DeviceType:    "Android",
+				},
+			},
+		},
+		{
+			name:    "Insert a new version and return init data when new version number is found",
+			wantErr: nil,
+			args: args{
+				deviceInfo: &models.Device{
+					ID:            "ccf2c998-992e-4db4-b395-22ac515d6c21",
+					VersionID:     "12da0a46-f94d-4ab3-8e3e-f4366d63d14d",
+					AppID:         "com.aerogear.testapp",
+					DeviceID:      "3ee2a7e7-25e4-465f-a998-c781b3cb9d46",
+					Version:       "2.1",
+					DeviceVersion: "8.1",
+					DeviceType:    "Android",
+				},
+			},
+			fields: fields{
+				app: &models.App{
+					ID:      uuid.New().String(),
+					AppID:   "com.aerogear.testapp",
+					AppName: "Test App",
+				},
+				version: &models.Version{
+					ID:                   "12da0a46-f94d-4ab3-8e3e-f4366d63d14d",
+					AppID:                "com.aerogear.testapp",
+					Version:              "2.0",
+					Disabled:             false,
+					DisabledMessage:      "",
+					NumOfCurrentInstalls: 1000,
+					NumOfAppLaunches:     10000,
+				},
+				device: &models.Device{
+					ID:            "ccf2c998-992e-4db4-b395-22ac515d6c21",
+					VersionID:     "12da0a46-f94d-4ab3-8e3e-f4366d63d14d",
+					AppID:         "com.aerogear.testapp",
+					DeviceID:      "3ee2a7e7-25e4-465f-a998-c781b3cb9d46",
+					Version:       "2.1",
+					DeviceVersion: "8.1",
+					DeviceType:    "Android",
+				},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			mockedRepository := &RepositoryMock{
```

Comment:
Use the mock repository for success and error on top of it. Check how is implemented the other tests where the mockRepo is passed as an argument. For some specific scenario which is valid just here, you can create a mock locally just implementing the functions which are used at this test.

Reply:
Hi @camilamacedo86 I think I need to have it in here. All of these functions are in use in this test.

Reply:
Hi @craicoverflow, the file already has the mocks as global for global use. See [here](https://github.com/aerogear/mobile-security-service/blob/master/pkg/web/apps/apps_service_test.go#L12) 
[here](https://github.com/aerogear/mobile-security-service/blob/master/pkg/web/apps/apps_service_test.go#L44) the global ones used for all tests implemented in this file. 

Also, look [here](https://github.com/aerogear/mobile-security-service/blob/master/pkg/web/apps/apps_service_test.go#L278) to get an example of how to do when the global ones are not enough and is required do more and specifics checks.

Reply:
@camilamacedo86 I cannot use the global mockRepository here. As you can see I am conditionally checking if the data from the test table matches that of the argument supplied in the mockRepository:

```go
GetAppByAppIDFunc: func(appID string) (*models.App, error) {
    if tt.fields.app == nil {
        return nil, models.ErrNotFound
    }

    return tt.fields.app, nil
},
```

The mock repositories you define are completely independent of the data from the test tables. They will not actually correctly mock the repository.

Reply:
I am afraid that I could not clarifies it for you. Let's move forward we can do a refactor in another PR. CLosig it.